docs/cellnet_architecture.rst

```diff
@@ -0,0 +1,213 @@
+.. _cellnet_architecture:
+
+FLARE CellNet Architecture
+--------------------------
+
+.. image:: resources/cellnet.png
+   :alt: FLARE CellNet Architecture
+
+
+Purpose and Scope
+#################
+
+CellNet is FLARE's unified communication layer that provides secure, scalable messaging between distributed federated
+learning components. It abstracts away network transport details and provides a consistent API for both small messages and
+large data transfers.
+
+Position in NVFLARE Architecture: CellNet sits between the application layer (Controllers, Executors, Admin commands) and
+the network transport layer (gRPC, TCP, HTTP drivers). All NVFLARE components communicate through CellNet, including:
+
+- **Server-to-client task distribution**
+- **Client-to-server result submission**
+- **Admin command execution**
+- **Cross-site auxiliary communication**
+- **Job deployment and management**
+
+Key Design Goals:
+#################
+
+- **Unified API**: Single interface for both small messages and large data streams
+- **Transport Agnostic**: Supports multiple network protocols (gRPC, TCP, HTTP)
+- **Hierarchical Addressing**: FQCN-based routing for multi-level cell hierarchies
+- **Secure Communication**: Built-in encryption and authentication
+- **Flow Control**: Automatic chunking and flow control for large transfers
+
+Three-Layer Architecture:
+#########################
+
+- **CoreCell**: Basic message routing, connection management, security
+- **StreamCell**: Large data streaming with chunking and flow control
+- **Cell**: High-level request/reply patterns with automatic channel detection
+
+Layered Cell Architecture
+#########################
+
+
+Three-Layer Design
+^^^^^^^^^^^^^^^^^^
+
+The CellNet architecture consists of three layers, each extending the previous:
+
+Layer 1: **CoreCell** - Basic Message Infrastructure
+provides fundamental messaging infrastructure:
+
+**Key Responsibilities**:
+
+- **Message Routing** - Routes messages to appropriate handlers based on channel/topic
```

Comment:
This should be called Message Handling. Message Routing is the capability to send message to cells not directly connected.

Reply:
changed